src/index.js

```diff
@@ -0,0 +1,172 @@
+import UploaderService from './service';
+
+const KILOBYTE = 1000,
+    MEGABYTE = 1024;
+
+class UploadController {
+
+    constructor({url, attachments = {}, maxFiles = 10, headers = {}, optionsObject = {}, updateCb, onLoad, maxFilesText = ''}) {
+        this.attachments = attachments;
+        this.maxFiles = maxFiles;
+        this.url = url;
+        this.optionsObject = optionsObject;
+        this.headers = headers;
+        this.fileCounter = Object.keys(this.attachments).length;
+        this.updateCb = updateCb;
+        this.onLoad = onLoad;
+        this.maxFilesText = maxFilesText;
+
+        this.onAttach = this.onAttach.bind(this);
+        this.onAbort = this.onAbort.bind(this);
+    }
+
+    static itemId(name) {
+        return `file_${name.match(/[a-zA-Z0-9_]/gi).join('')}`;
+    }
+
+    static maxFilesReached(fileCount, maxFiles) {
+        return fileCount >= maxFiles;
+    }
+
+    static stripNameFromExtension(name) {
+
+        const lastDot = name.lastIndexOf('.'),
+            nameLength = name.length,
+            hasExtension = (nameLength - lastDot) <= 5;
+
+        let extension = '';
+
+        if (hasExtension) {
+            extension = name.slice(lastDot);
+            name = name.split(extension)[0];
+        }
+
+        return {
+            name,
+            extension
+        };
+    }
+
+    static prettyFileSize(size) {
+        const sizeInKB = Math.round(size / KILOBYTE);
+        let unit = 'KB';
+
+        if (sizeInKB < MEGABYTE) {
+            return sizeInKB + unit;
+        } else {
+            unit = 'MB';
+            const sizeInMB = (sizeInKB / MEGABYTE).toFixed(2);
+
+            return sizeInMB + unit;
+        }
+    }
+
+    static getFileReady(file) {
+
+        const fileMeta = file,
+            fullName = UploadController.stripNameFromExtension(fileMeta.name),
+            fileSize = UploadController.prettyFileSize(fileMeta.size);
+
+        fileMeta.fileName = fullName.name;
+        fileMeta.extension = fullName.extension;
+        fileMeta.fileSize = fileSize;
+        fileMeta.progress = 0;
+
+        return fileMeta;
+    }
+
+    update() {
+        if (!this.updateCb) {
+            return;
+        }
+
+        const files = {};
+
+        for (const attachment in this.attachments) {
+            if (this.attachments[attachment].hasOwnProperty('file')) {
+                files[attachment] = this.attachments[attachment].file;
+            }
+        }
+
+        this.updateCb(files);
```

Comment:
not filtering, just giving back the file data, without the file objects. I believe the api users should not bother with it.
I can, instead, save both in two different objects, and then, just hand them back instead of iterating over and over again each time. Would that be better?